test/test_transformers.py

```diff
@@ -849,6 +849,42 @@ def test_encoder_is_causal(self):
 
         self.assertEqual(masked_output, is_causal_output)
 
+    @onlyCUDA
+    def test_math_backend_high_precision(self):
```

Comment:
the test coverage I'm looking for it just verifying that setting the flag does indeed turn on low precision mode. This might be as easy as comparing the differences between:
1. flag off SDPA math vs. the "gold standard"
2. flag on SDPA math vs. the "gold standard"

the difference for 1 should be substantially < 2 if the flag is working correctly

I see you're using flash as the gold standard, but is it better to use float64 math? If flash is preferred, we'll need to decorate this test so it only runs on platforms where flash is supported e.g.
https://github.com/pytorch/pytorch/blob/599735415124d78f2b1766681da1547a62681c8d/test/test_transformers.py#L1393-L1395

Reply:
Update the flash backend with float64 math as you suggested.

Reply:
We used `sdp_math_low_prec_out` for 1, and `sdp_math_high_prec_out` for 2.

The test shows that 2 can pass the 1st torch.testing.assert_close; but 1 will fail.

Reply:
okay looks good, thanks for clarifying! FYI there's some sugar you can use when you expect an error:
```python
with self.assertRaisesRegex(AssertionError, "Tensor-likes are not close"):
    ...
```